internal/controller/deployment.go

```diff
@@ -121,6 +118,10 @@ func ConstructEnvForApp(ctx context.Context, app *spinv1alpha1.SpinApp) []corev1
 		envs[idx] = env
 	}
 
+	// Adding the OpenTelemetry params
+	otelVars := []corev1.EnvVar{{Name: "OTEL_EXPORTER_OTLP_ENDPOINT", Value: otel.Endpoint}}
+	envs = append(envs, otelVars...)
```

Comment:
```suggestion
	if otel.Endpoint != "" {
	  envs = append(envs, corev1.EnvVar{Name: "OTEL_EXPORTER_OTLP_ENDPOINT", Value: otel.Endpoint})
	}
```

no need to construct a second list - can just directly add things to envs.